examples/animation/animated_histogram.py

```diff
@@ -3,89 +3,64 @@
-# * We need 3 ``LINETO``'s, which tell Matplotlib to draw lines from
-#   vertex 1 to vertex 2, v2 to v3, and v3 to v4.
-# * We then need one ``CLOSEPOLY`` which tells Matplotlib to draw a line from
-#   the v4 to our initial vertex (the ``MOVETO`` vertex), in order to close the
-#   polygon.
-#
-# .. note::
-#
-#   The vertex for ``CLOSEPOLY`` is ignored, but we still need a placeholder
-#   in the ``verts`` array to keep the codes aligned with the vertices.
-nverts = nrects * (1 + 3 + 1)
-verts = np.zeros((nverts, 2))
-codes = np.full(nverts, path.Path.LINETO)
-codes[0::5] = path.Path.MOVETO
-codes[4::5] = path.Path.CLOSEPOLY
-verts[0::5, 0] = left
-verts[0::5, 1] = bottom
-verts[1::5, 0] = left
-verts[1::5, 1] = top
-verts[2::5, 0] = right
-verts[2::5, 1] = top
-verts[3::5, 0] = right
-verts[3::5, 1] = bottom
+n, _ = np.histogram(data, HIST_BINS)
 
 ###############################################################################
 # To animate the histogram, we need an ``animate`` function, which generates
-# a random set of numbers and updates the locations of the vertices for the
-# histogram (in this case, only the heights of each rectangle). ``patch`` will
-# eventually be a `.Patch` object.
-patch = None
+# a random set of numbers and updates the heights of rectanges. We utilize a
+# python closure to track an instance of `.BarContainer` whose `.Rectangle`
+# patches we shall update.
 
 
-def animate(i):
-    # simulate new data coming in
-    data = np.random.randn(1000)
-    n, bins = np.histogram(data, 100)
-    top = bottom + n
-    verts[1::5, 1] = top
-    verts[2::5, 1] = top
-    return [patch, ]
+def prepare_animation(bar_container):
+
+    def animate(frame_number):
+        # simulate new data coming in
+        data = np.random.randn(1000)
+        n, _ = np.histogram(data, HIST_BINS)
+        for count, rect in zip(n, bar_container.patches):
+            rect.set_height(count)
+        return bar_container.patches
+    return animate
 
 ###############################################################################
-# And now we build the `.Path` and `.Patch` instances for the histogram using
-# our vertices and codes. We add the patch to the `~.axes.Axes` instance, and
-# setup the `.FuncAnimation` with our ``animate`` function.
-fig, ax = plt.subplots()
-barpath = path.Path(verts, codes)
-patch = patches.PathPatch(
-    barpath, facecolor='green', edgecolor='yellow', alpha=0.5)
-ax.add_patch(patch)
+# Using :func:`~matplotlib.pyplot.hist` allows us to get an instance of
+# `.BarContainer`, which is a collection of `.Rectangle` instances. Calling
+# ``prepare_animation`` will define ``animate`` function working with supplied
+# `.BarContainer`, all this is used to setup `.FuncAnimation`.
 
-ax.set_xlim(left[0], right[-1])
-ax.set_ylim(bottom.min(), top.max())
+fig, ax = plt.subplots()
+_, _, bar_container = ax.hist(
+    data,
+    HIST_BINS,
+    lw=1,
+    ec="yellow",
+    fc="green",
+    alpha=0.5
+)
```

Comment:
```suggestion
_, _, bar_container = ax.hist(data, HIST_BINS, lw=1,
                              ec="yellow", fc="green", alpha=0.5)
```

While we have no formal specification on the wrapping style, we're currently not using one-parameter-per-line anywhere in the code or docs. Therefore, I'd like to keep this compact.

Reply:
I know, I have noticed the communication regarding the formatting. I personally prefer the long option, mostly because I am really used to read it. But for documentation/example purpose I find the compact form nice, shifting some attention from code itself to texts and figures.